cmd/crdvalidator/handlers/crd.go

```diff
@@ -57,6 +59,11 @@ func (cv *CrdValidator) Handle(ctx context.Context, req admission.Request) admis
 		return admission.Errored(http.StatusBadRequest, fmt.Errorf("%s: %w", message, err))
 	}
 
+	// Only run validation on CRDs with the ValidateAnnotation set to true
+	if val, ok := incomingCrd.GetAnnotations()[ValidateAnnotation]; !ok || val != "true" {
```

Comment:
I was thinking, instead of watching all CRD events, and then filtering out the ones for CRDs that don't have the annotation set, we could build a predicate into the webhook's cache such that it only receives events for CRDs that have this annotation/label set. That approach is more scalable (since we don't see events we don't care about) and intuitive, since there is no filtering in the webhook code itself.

Reply:
Good call, that makes absolute sense. Let me iterate on this and see what I can do to that end.

Reply:
Implemented 🔥 Great suggestion.